apps/arweave/src/ar_data_sync.erl

```diff
@@ -2820,6 +2827,11 @@ process_valid_fetched_chunk(ChunkArgs, Args, State) ->
 					%% The chunk has been synced by another job already.
 					decrement_chunk_cache_size(),
 					{noreply, State};
+				false when Packing =/= unpacked ->
+					true = AbsoluteEndOffset == AbsoluteTXStartOffset + ChunkEndOffset,
+					pack_and_store_chunk({DataRoot, AbsoluteEndOffset, TXPath, TXRoot,
+							DataPath, Packing, ChunkEndOffset, ChunkSize, Chunk,
+							none, none, none}, State);
```

Comment:
Do you know why we need this extra case? Looks like the only difference is passin gin `none` instead of `UnpackedChunk`?

Reply:
This is a dirty fix for the problem, I will refactor this to make it more clear.
But the reason for having this clause here is the wrong naming of the variables inside the `ChunkArgs`. It actually can hold `Packing` + `PackedChunk`, just the original one. So we need to manually set `UnpackedChunk` to `none` here.